File: 66-plus-one/66-plus-one.py
class Solution(object):
    def plusOne(self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """
        carry = 0
        for i in range(len(digits)):
            if i == 0 :
                temp = (digits[len(digits)-i-1] + 1 + carry)%10
                carry = (digits[len(digits)-i-1] + 1 + carry)/10
                digits[len(digits)-i-1] = temp
            else:
                temp = (digits[len(digits)-i-1] + carry)%10
                carry = (digits[len(digits)-i-1] + carry)/10
                digits[len(digits)-i-1] = temp
            
        if carry != 0 :
            digits = [1] + digits
            
        return digits

File: 66-plus-one/66-plus-one.java
class Solution {
    public int[] plusOne(int[] digits) {
        int carry=1;
        for(int n=digits.length-1;n>=0;n--){
            digits[n]=digits[n]+carry;
            carry=digits[n]/10;
            if(digits[n]>9){
                digits[n]%=10;
            }
        }
        if(carry>0){
            int[] result=new int[digits.length+1];
            result[0]=1;
            System.arraycopy(digits,0,result,1,digits.length);
            return result;
        }
        return digits;
    }
}